docs/src/modules/components/AppFrame.js

```diff
@@ -237,7 +237,7 @@ function AppFrame(props) {
               <ExpandMoreIcon fontSize="small" />
             </Button>
           </Tooltip>
-          <NoSsr>
+          <NoSsr defer>
```

Comment:
This is also something I would suggest for a breaking change: default to `defer` i.e. `defer` -> `noDefer`. The recommendation for client-only components from core members is to use `useEffect`.

Reply:
It see pros and cons with both default values. I think that it depends on why a developer would use the component for.

Right know, we advertise it as a way to prune the rendering server side. In this context, it's important to keep a single commit, hence defer = false by default.

I think that we could imagine the multi commit use case of defer = true as a new component. In the current React infrastructure, it can only be achieved without rendering on the server. It's unsure how this constraint will hold in the future.

What do you think?

Reply:
> by default to avoid layout jumps.

This is not true though. Between html render and react hydration you have different layout. You will always have a layout jump with NoSsr.

Reply:
Yes, sorry, I should have been more precise: between two renders after hydration.

Reply:
In that case nothing changes? The state of the component only changes on the initial mount.

Reply:
I believe it makes a difference, for instance, when using Next.js and rendering a new page.

- page A
- page B with this simplified React tree: X > NoSsr > Y

If you load page A, and navigate to page B, client-side, you will notice a difference with or without the defer prop. If `defer`, page B will render in two steps, if not `defer`, page B will render in a single commit.

Reply:
Right. We don't actually care abouting mounted state but hydration state.